hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/impl/BurnPrecompile.java

```diff
@@ -159,4 +172,20 @@ public static BurnWrapper decodeBurn(final Bytes input) {
                     tokenID, Arrays.stream(serialNumbers).boxed().toList());
         }
     }
+
+    public static BurnWrapper decodeBurnV2(final Bytes input) {
+        final Tuple decodedArguments =
+                decodeFunctionCall(input, BURN_TOKEN_SELECTOR_V2, BURN_TOKEN_DECODER);
```

Comment:
You can add a common private method for the two decode methods, that constructs `BurnWrapper` from a `Tuple`.